### ISSUE TITLE
[ao/float8] Discussion of the current Float8 recipe

### ISSUE THREAD

**Issue by yanbing-j** [pytorch/ao#972]
Now in torchao/float8, only `Float8Linear` and operators in `FLOAT8_OPS_TABLE` in `torchao/float8/float8_ops.py` are supported Float8, that is also called the `recipe` of float8 in torchao.

I have a few questions about the `recipe` of Float8 in torchao:

- What is the test scope of the current recipe and Float8 scheme? Only for Llama2 70b and Llama3 8b?
- Will other operators extend to Float8 implementation, e.g, LayerNorm? Or Is there any plan about this?
- Are the current recipe enough to guarantee the accuracy of real models, like Llama2 70b and Llama3 8b?

Thanks!

**Comment by yanbing-j** [comment]
@vkuzo

Thanks for the detailed comments. For the accuracy, I find the codes in ao/float8 is lack of related accuracy tests. Do you mean the recipe of only supporting torch.nn.Linear is same as https://arxiv.org/pdf/2209.05433, and the accuracy should be same as Section 4? And do you have any tests in your local and with ao/float8?

Thanks!

**Comment by vkuzo** [comment]
> Do you mean the recipe of only supporting torch.nn.Linear is same as https://arxiv.org/pdf/2209.05433, and the accuracy should be same as Section 4?

Yes, that's right!

> Do you have any tests in your local and with ao/float8?

So far for float8 we have been focusing on large scale pretraining, the largest scale we have reported public accuracy results for is 128 H100s on pretraining LLaMa 3 70B: https://dev-discuss.pytorch.org/t/enabling-float8-all-gather-in-fsdp2/2359.  If you share some more context about what you are interested in, happy to help further!

**Comment by yanbing-j** [comment]
Thanks for the confirmation!
